scatter-gather/scatter-gather.bal

```diff
@@ -0,0 +1,55 @@
+import ballerina/http;
+
+type CarRental record {|
+    string carType;
+    string price;
+    string resultLink;
+|};
+
+service /api/v1 on new http:Listener(8080) {
+    
+    resource function get vehicles(string dropOffDateTime, string dropOffLocation, string pickupDateTime, string pickupLocation) returns CarRental[]|error {
+        return requestRentalCar(dropOffDateTime, dropOffLocation, pickupDateTime, pickupLocation);
+    }
+}
+
+function requestRentalCar(string dropOffDateTime, string dropOffLocation, string pickupDateTime, string pickupLocation) returns CarRental[]|error {
+    
+    worker hotwire returns CarRental[]|error {
+        http:Client hotwireEP = check new ("http://api.hotwire.com.balmock.io");
+        http:Response response = check hotwireEP -> /v1/search/car.get(pickup=pickupLocation, dest=dropOffLocation, startDate=pickupDateTime, endDate=dropOffDateTime);
+        return transformHotwireResponse(check response.getXmlPayload());
+    }
+
+    worker avis returns CarRental[]|error {
+        http:Client avisEP = check new ("http://api.avis.com.balmock.io");
+        http:Response response = check avisEP -> /cars/catalog/v1/vehicles/rates.get(brand="Avis", country_code="US", dropOff_date=dropOffDateTime, dropoff_location=dropOffLocation, pickup_date=pickupDateTime, pickup_location=pickupLocation);
+        return transformAvisResponse(check response.getJsonPayload());
+    }
+
+    CarRental[] result = [];
+    map<CarRental[]|error> mapResult = wait {hotwire, avis};
+    from CarRental[]|error carRental in mapResult
+        where carRental is CarRental[]
+        do {
+            result.push(...carRental);
+        };
+    return result;
```

Comment:
Let's `return wait {hotwire, avis}` and change the return type accordingly. This is to reduce the length of the sample.

Reply:
Even we return the `wait {hotwire, avis}`, later we have to process to update the final response correctly.

Reply:
I don't think scatter gather patterns says you have to pick a single response, replying all in a map is still valid.

Reply:
Your argument is valid. But when a developer runs this sample he should see something nice. Having a array is better than having a map which contains arrays as values.

Reply:
Here the map output will be in this structure `{"avis": CarRental[], "hotwire": CarRental[] }`. This way I said returning an array would be better.

Reply:
Is that structure bad?

Reply:
In a real scenario the client will only interested in the array result. If use the `map.values()` the result will be a nested array.

Reply:
If we go with the current approach following will be the output.
```json
[
  {
    "carType": "Chevy Uplander",
    "price": "USD 229.84",
    "resultLink": "http://siva.com/quote?id=95900"
  },
  {
    "carType": "Hyundai Accent",
    "price": "USD 214.40",
    "resultLink": "http://hotline.com/quote/1234"
  },
  {
    "carType": "Ford Freestar",
    "price": "USD 242.13",
    "resultLink": "http://hotline.com/quote/1257"
  }
]
```

Reply:
Response after your suggestion, 
```json
{
  "siva": [
    {
      "carType": "Chevy Uplander",
      "price": "USD 229.84",
      "resultLink": "http://siva.com/quote?id=95900"
    }
  ],
  "hotline": [
    {
      "carType": "Hyundai Accent",
      "price": "USD 214.40",
      "resultLink": "http://hotline.com/quote/1234"
    },
    {
      "carType": "Ford Freestar",
      "price": "USD 242.13",
      "resultLink": "http://hotline.com/quote/1257"
    }
  ]
}
```